src/features/Licenses/LicensesFilters/index.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { Col, Form } from '@edx/paragon';
+import { Select, Button } from 'react-paragon-topaz';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { initialPage } from 'features/constants';
+import { fetchCoursesData } from 'features/Courses/data/thunks';
+import { updateFilters, updateCurrentPage } from 'features/Licenses/data/slice';
+
+import { fetchLicensesData } from '../data';
+
+const LicensesFilters = ({ resetPagination }) => {
+  const dispatch = useDispatch();
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const courses = useSelector((state) => state.courses.table.data);
+  const licenses = useSelector((state) => state.licenses.table.data);
+
+  const licenseData = useRef(null);
+  const [courseOptions, setCourseOptions] = useState([]);
+  const [licenseOptions, setLicenseOptions] = useState([]);
+  const [courseSelected, setCourseSelected] = useState(null);
+  const [licenseSelected, setLicenseSelected] = useState(null);
+
+  const isButtonDisabled = courseSelected === null && licenseSelected === null;
+
+  const handleSelectFilters = async (e) => {
+    e.preventDefault();
+
+    if (isButtonDisabled) {
+      return;
+    }
+
+    const form = e.target;
+    const formData = new FormData(form);
+    const formJson = Object.fromEntries(formData.entries());
+    const urlParams = Object.entries(formJson)
+      .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
+      .join('&');
+
+    try {
+      dispatch(updateFilters(formJson));
+      dispatch(updateCurrentPage(initialPage));
+      licenseData.current = licenseOptions;
+      dispatch(fetchLicensesData(institution.id, initialPage, urlParams));
+      setLicenseOptions(licenseData.current);
+    } catch (error) {
+      logError(error);
+    }
+  };
+
+  const handleCleanFilters = () => {
+    dispatch(fetchLicensesData(institution.id, initialPage));
+    resetPagination();
+    setLicenseSelected(null);
+    setCourseSelected(null);
+    dispatch(updateFilters({}));
+  };
+
+  useEffect(() => {
+    if (courses.length > 0) {
+      const options = courses.map(course => ({
+        ...course,
+        label: course.masterCourseName,
+        value: course.masterCourseName,
+      }));
+      setCourseOptions(options);
+    }
+
+    if (licenses.length > 0 && licenseData.current === null) {
```

Comment:
Why is necessary validate the licenses ref?

Reply:
That was for to avoid a bug when you select a license